cachito/web/api_v1.py

```diff
@@ -65,29 +64,21 @@ def download_archive(request_id):
         raise ValidationError(
             'The request must be in the "complete" state before downloading the archive')
 
-    cachito_shared_dir = flask.current_app.config['CACHITO_SHARED_DIR']
-    wait_timeout = flask.current_app.config['CACHITO_WAIT_TIMEOUT']
-
-    with tempfile.TemporaryDirectory(prefix='cachito-', dir=cachito_shared_dir) as temp_dir:
-        # Although the cachito_shared_dir volume is required to be the same between celery
-        # workers and the API, they may be mounted at different locations. Use relative
-        # paths to agree on data location within the shared volume.
-        relative_temp_dir = os.path.basename(temp_dir)
-        relative_deps_path = os.path.join(relative_temp_dir, 'deps')
-        relative_bundle_archive_path = os.path.join(relative_temp_dir, 'bundle.tar.gz')
-        absolute_bundle_archive_path = os.path.join(
-            cachito_shared_dir, relative_bundle_archive_path)
-
-        # Chain tasks
-        chain_result = chain(
-            tasks.fetch_app_source.s(request.repo, request.ref),
-            tasks.fetch_gomod_source.s(copy_cache_to=relative_deps_path),
-            tasks.assemble_source_code_archive.s(
-                deps_path=relative_deps_path, bundle_archive_path=relative_bundle_archive_path)
-        ).delay()
-        chain_result.wait(timeout=wait_timeout)
-
-        return flask.send_file(absolute_bundle_archive_path, mimetype='application/gzip')
+    cachito_archives_dir = flask.current_app.config['CACHITO_ARCHIVES_DIR']
+    bundle_archive_path = os.path.join(
+        cachito_archives_dir, 'cachito_bundles', f'{request_id}.tar.gz')
+    if not os.path.exists(bundle_archive_path):
+        flask.current_app.logger.error(
+            'The bundle archive at %s for request %d doesn\'t exist',
+            bundle_archive_path, request_id,
+        )
+        raise InternalServerError()
```

Comment:
Would `Not Found` make more sense here? I see your logic with InternalServerError in case the bundle should always be there for complete request.

Reply:
Yes, it should always exist. If it doesn't, it's an issue such as the volume not being mounted or it got accidentally deleted.